clustering/web/infinispan/src/main/java/org/wildfly/clustering/web/infinispan/SessionIdentifierSerializer.java

```diff
@@ -25,25 +25,26 @@
 import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
+import java.security.PrivilegedAction;
 import java.util.ServiceLoader;
 
 import org.wildfly.clustering.marshalling.spi.Serializer;
 import org.wildfly.clustering.web.IdentifierSerializerProvider;
+import org.wildfly.security.manager.WildFlySecurityManager;
 
 /**
  * @author Paul Ferraro
  */
 public enum SessionIdentifierSerializer implements Serializer<String> {
     INSTANCE;
 
-    private final Serializer<String> serializer = load(IdentifierSerializerProvider.class.getClassLoader());
-
-    private static final Serializer<String> load(ClassLoader loader) {
-        for (IdentifierSerializerProvider provider : ServiceLoader.load(IdentifierSerializerProvider.class, loader)) {
+    private final Serializer<String> serializer = WildFlySecurityManager.doUnchecked((PrivilegedAction<Serializer<String>>) () -> {
+        //noinspection LoopStatementThatDoesntLoop
```

Comment:
Instead of ignoring codestyle warning, so long as you're touching this, you can convert the Iterable to an iterator, and throw the exception if hasNext() is false.

Reply:
Didn't want the thread on your style. Changed now.